### PR TITLE
improvments to library UI status reporting

### PR DESCRIPTION
**Pull request by EladKeyshawn**

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 1]
```diff
@@ -45,259 +47,296 @@
...
+    private ZimManageActivity zimManageActivity;
+    CoordinatorLayout mainLayout;
+    private Activity faActivity;
+    private boolean hasArtificiallyPaused;
 
     @Override
-    public int getCount() {
-      return mData.size();
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        faActivity = super.getActivity();
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: Wrong indenting.

File: app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java [hunk 2]
```diff
@@ -110,25 +120,26 @@ public static String toHumanReadableTime(int seconds) {
 
     if (Math.round(seconds / DAYS) > 0)
       return String.format(Locale.getDefault(), "%d %s %s", Math.round(seconds / DAYS),
-          KiwixApplication.getInstance().getResources().getString(R.string.time_day),
-          KiwixApplication.getInstance().getResources().getString(R.string.time_left));
+              KiwixApplication.getInstance().getResources().getString(R.string.time_day),
```
Comment (mhutti1) [inline app/src/main/java/org/kiwix/kiwixmobile/downloader/DownloadFragment.java]: This should be 4 spaces like before.

**Comment by mhutti1** [comment]
@EladKeyshawn What does this do?

**Comment by EladKeyshawn** [comment]
@mhutti1 better ui to logic sync with available items (books), as well as in the downloads fragment gets updated if download is stopped from notifications and some code redundancy clean up

**Comment by kelson42** [comment]
@mhutti1 @EladKeyshawn Could someone please rebase this against master so we can merge it? Please avoid to merge master to a feature branch, rebase instead. Otherwise, the git history will become quite messy.
@mhutti1 If you review a PR and its OK for you, can you please merge it directly? This would simplify the workflow and avoid this kind of situation.

**Comment by mhutti1** [comment]
I would prefer the creator of a pull request performs a merge themselves after a review and passing tests as this prevents unfinished code getting committed.

**Comment by kelson42** [comment]
To resync feature branches, please use rebase instead of merge.

**Comment by mhutti1** [comment]
As there was a merge earlier in this history a rebase is not possible online. It might be possible manually but I am not sure.

**Comment by kelson42** [comment]
@mhutti1 the github UI provides only a really limited set of tools. All these operations should be better done on the command line. And "yes", it might not always be straigt-forward... but usually, if things are done properly (also in the past), then everything works quite smoothly :)

**Pull request merged**